### ISSUE TITLE
Re-render options after @search-change 

### ISSUE THREAD

**Issue by rinterliche** [riophae/vue-treeselect#18]
On my case, what a I need is to call an Ajax call after the `@search-change` redefining the options then. But, it seem that setting the options remotely is not permitted.

```
<template lang="pug">
    treeselect(
        v-model="value"
        :options="options"
        @search-change="searchChange"
    )
</template>

<script>
    export default {
        options: [],
        value: null,
        methods: {
            searchChange () {
                 this.doAjaxRequest().then((body) => {
                     this.options = body.data
                 })
            }
        }
    }
</script>
```

Doing this way causes the options to be emptied. Am I doing something wrongly or do I have to  implement it?

Example:
http://sagalbot.github.io/vue-select/docs/Advanced/Ajax.html

**Comment by chenEdgar** [comment]
when have a look in the source code, you will find that each `vue-treeselect` component will initicial the options in create life circle by calling `initialize` method. So if you want to rerender options using data by async, you need to use `loadRootOptions` or `loadChildrenOptions`. another method is not render it until you wait a ajax return.

**Comment by sandeep240288** [comment]
How can I call an ajax request to a backend? Please provide a code demo. which will help a lot

**Comment by rinterliche** [comment]
@chenEdgar `loadRootOptions` and `loadChildrenOptions` are passed as props. I need a way for the search input to execute an certain callback on change, changing the options.

**Comment by chenEdgar** [comment]
I see. you did it in your initial comment demo link.

**Comment by sandeep240288** [comment]
I want options to be reevaluated from the database once a property change in parent div.

`the code is as follows
<div class="control">
          <label for="standard" class="label">Select Standard:</label>
          <select type="text"   name="standard" class="form-control" v-model="form.standard" @change="onselect">
            <option  v-for="std in stds" :value="std.id">  {{std.id +' - '+ std.description}}</option>
          </select>
          <span class="help text-danger" v-if="form.errors.has('standard')" v-text="form.errors.get('standard')"></span>
      </div>

///`
once an element is selected in above dropdown, it should trigger the new set of options based on the id of a selected element in this option.

**Comment by chenEdgar** [comment]
@sandeep240288 sorry, I can't understand you.  can you show us your code or reproducing a demo?

**Comment by riophae** [comment]
@rinterliche @sandeep240288 Sorry for getting back to you so late! Currently vue-treeselect doesn't support that feature, but it has been planned though.

**Comment by riophae** [comment]
Hi! Async searching is finally landed in v0.0.31. Check out: https://vue-treeselect.js.org/#async-searching
